core/shared/src/main/scala/fs2/internal/Algebra.scala

```diff
@@ -52,6 +55,19 @@ private[fs2] object Algebra {
       })
     }
 
+  /**
+    * Like `scope` but allows this scope to be interrupted.
+    * Note that this may fail with `Interrupted` when interruption occured
+    */
+  private[fs2] def interruptScope[F[_], O, R](pull: FreeC[Algebra[F,O,?],R])(implicit effect: Effect[F], ec: ExecutionContext): FreeC[Algebra[F,O,?],R] = {
```

Comment:
Can we define both this and `scope` in terms of a new method that has the bulk of this body?

Reply:
ok